Dashboard fixes

tremainebuchanan opened the pull request:
Fixes done to prepare app for production.
Uncommitted lines that prevented the Input list item from showing.
Ensured that only the admin user can add a Farm Store.
Ensured that a user will not select a "UNIT" when creating a demand or supply.

tremainebuchanan committed:
Added new unit type

tremainebuchanan committed:
Fixes done to prepare app for production.
Uncommented lines that prevented the Input list item from showing.
Ensured that only the admin user can add a Farm Store.
Ensured that a user will not select a "UNIT" when creating a demand
or supply.

matjames007 commented:
Ok, I am confused.  Could you clarify what you meant by:

> "Ensured that a user will not select a "UNIT" when creating a demand or supply."

What is UNIT? and why is this a problem?

tremainebuchanan commented:
When we are creating a input for a farm supply store, we wanted to prevent the user from selecting the incorrect unit of measure. For example if a user is creating a shovel we wouldn't want them to save the farm input as $100 shovel per lb. Ideally we would want the input to be saved as $100 shovel per unit.

matjames007 commented:
I see.  I understand now.  I don't think this is the correct approach.  Understand that the logic you have is currently accessible by the web application and there will be issues when it comes to other applications connecting to the api.

Consider having a type defined on the unit.  For demands and commodities the type being used in the web application would be "mass/weight".  For inputs etc it could use all types a type that defines arbitrary unit values.

Matching of demands and supplies can only occur using units with mass/weight values.

tremainebuchanan committed:
Added unit_class to Unit Schema

tremainebuchanan committed:
Added unit class to units.
Unit Type drop downs now render appropriate unit
types.
The matching of Demands by a unit class was not done.
The rational being that since we are pulling and rendering
the correct unit class for a demand/supply creation,
the correct unit class will always be stored within a demand/supply.

tremainebuchanan committed:
Reverted changes made to Unit Schema
The un_unit_name has been returned to a unique status.
Units retrieved and rendered in on Farmer's and Buyer's
profile page will only be of a unit class "Weight".
Units retrieved and rendered on the input dialog
will be units of all un_unit_class type.

tremainebuchanan committed:
Added unique attribute to un_unit_name
